File: Python/strings.py
# Initialize a sample string
sample_string = "Hello, World! This is a sample string."

# 1. Length of the string
length = len(sample_string)
print(f"1. Length of the string: {length}")

# 2. Convert to uppercase
uppercase_string = sample_string.upper()
print(f"2. Uppercase: {uppercase_string}")

# 3. Convert to lowercase
lowercase_string = sample_string.lower()
print(f"3. Lowercase: {lowercase_string}")

# 4. Capitalize the first letter
capitalized_string = sample_string.capitalize()
print(f"4. Capitalized: {capitalized_string}")

# 5. Count occurrences of a substring
substring = "is"
count = sample_string.count(substring)
print(f"5. Count of '{substring}' in the string: {count}")

# 6. Find the index of a substring (first occurrence)
substring = "World"
index = sample_string.find(substring)
print(f"6. Index of '{substring}': {index}")

# 7. Replace a substring
old_substring = "sample"
new_substring = "modified"
replaced_string = sample_string.replace(old_substring, new_substring)
print(f"7. Replace '{old_substring}' with '{new_substring}': {replaced_string}")

# 8. Check if the string starts with a prefix
prefix = "Hello"
starts_with = sample_string.startswith(prefix)
print(f"8. Starts with '{prefix}': {starts_with}")

# 9. Check if the string ends with a suffix
suffix = "string."
ends_with = sample_string.endswith(suffix)
print(f"9. Ends with '{suffix}': {ends_with}")

# 10. Split the string into a list of substrings
split_string = sample_string.split()
print(f"10. Split the string into words: {split_string}")

# 11. Join a list of substrings into a string
joined_string = " ".join(split_string)
print(f"11. Join the list into a string: {joined_string}")

# 12. Strip leading and trailing whitespace
whitespace_string = "   This has whitespace.   "
stripped_string = whitespace_string.strip()
print(f"12. Stripped: '{stripped_string}'")

# 13. Check if the string is numeric
numeric_string = "12345"
is_numeric = numeric_string.isnumeric()
print(f"13. Is numeric? {is_numeric}")

# 14. Check if the string is alphabetic
alphabetic_string = "abcdef"
is_alphabetic = alphabetic_string.isalpha()
print(f"14. Is alphabetic? {is_alphabetic}")

# 15. Check if the string is alphanumeric
alphanumeric_string = "abc123"
is_alphanumeric = alphanumeric_string.isalnum()
print(f"15. Is alphanumeric? {is_alphanumeric}")

File: TypeScript/Strings.ts
// Sample string
const sampleString = "Hello, world! This is a sample string.";

// Using string methods directly
console.log("Original String:", sampleString);
console.log("Length:", sampleString.length);
console.log("Uppercase:", sampleString.toUpperCase());
console.log("Lowercase:", sampleString.toLowerCase());
console.log("Substring from index 7 to 12:", sampleString.substring(7, 12));
console.log("Index of 'world':", sampleString.indexOf("world"));
console.log("Replace 'sample' with 'modified':", sampleString.replace("sample", "modified"));
console.log("Split by space:", sampleString.split(" "));
console.log("Trimmed:", sampleString.trim());
console.log("Concatenated with ' - appended string':", sampleString.concat(" - appended string"));
console.log("Includes 'world'?", sampleString.includes("world"));
console.log("Starts with 'Hello'?", sampleString.startsWith("Hello"));
console.log("Ends with 'string'?", sampleString.endsWith("string"));
console.log("Character at index 4:", sampleString.charAt(4));
console.log("Code unit value at index 4:", sampleString.charCodeAt(4));
console.log("Last index of 'i':", sampleString.lastIndexOf("i"));

// ======= Functional Usage =============
// Function to count the occurrences of a substring in a string
function countOccurrences(str: string, subStr: string): number {
    return str.split(subStr).length - 1;
}

// Function to reverse a string
function reverseString(str: string): string {
    return str.split('').reverse().join('');
}

// Function to check if a string is a palindrome
function isPalindrome(str: string): boolean {
    const reversedStr = str.split('').reverse().join('');
    return str === reversedStr;
}

// Function to truncate a string
function truncateString(str: string, maxLength: number): string {
    if (str.length > maxLength) {
        return str.slice(0, maxLength) + '...';
    }
    return str;
}

// Function to capitalize the first letter of each word in a string
function capitalizeWords(str: string): string {
    return str.replace(/\b\w/g, char => char.toUpperCase());
}

// Sample usage
const sample = "hello world";
console.log("Original String:", sample);
console.log("Number of 'l' occurrences:", countOccurrences(sample, 'l'));
console.log("Reversed String:", reverseString(sample));
console.log("Is palindrome?", isPalindrome(sample));
console.log("Truncated String (max length 5):", truncateString(sample, 5));
console.log("Capitalized Words:", capitalizeWords(sample));
